http.go

```diff
@@ -256,51 +259,120 @@ func Get(w http.ResponseWriter, req *http.Request, store mdata.Store, metricInde
 
 	reqs := make([]Req, 0)
 	for _, target := range targets {
-		var consolidateBy string
-		id := target
-		// yes, i am aware of the arguably grossness of the below.
-		// however, it is solid based on the documented allowed input format.
-		// once we need to support several functions, we can implement
-		// a proper expression parser
-		if strings.HasPrefix(target, "consolidateBy(") {
-			t := target
-			if t[len(t)-2:] != "')" || (!strings.Contains(t, ",'") && !strings.Contains(t, ", '")) || strings.Count(t, "'") != 2 {
-				http.Error(w, "target parse error", http.StatusBadRequest)
-				return
-			}
-			consolidateBy = target[strings.Index(target, "'")+1 : strings.LastIndex(target, "'")]
-			id = target[strings.Index(target, "(")+1 : strings.LastIndex(target, ",")]
+		id, consolidateBy, err := parseTarget(target)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
 		}
 
+		type locatedDef struct {
+			def schema.MetricDefinition
+			loc string
+		}
+
+		locatedDefs := make(map[string]locatedDef)
+
 		if legacy {
 			nodes, err := metricIndex.Find(org, id)
 			if err != nil {
 				http.Error(w, err.Error(), http.StatusBadRequest)
 				return
 			}
-			if len(nodes) == 0 {
-				http.Error(w, errMetricNotFound.Error(), http.StatusBadRequest)
-				return
-			}
 			for _, node := range nodes {
 				for _, def := range node.Defs {
-					consolidator, err := consolidation.GetConsolidator(&def, consolidateBy)
-					if err != nil {
-						http.Error(w, err.Error(), http.StatusBadRequest)
-						return
+					locatedDefs[def.Id] = locatedDef{def, "local"}
+				}
+			}
+
+			for _, inst := range otherNodes {
+				res, err := http.Get(inst)
+				http.PostForm("http://%s/index/find", url.Values{"target": targets})
```

Comment:
this looks wrong.  You are making a GET request for "inst", then posting a form to ??